docs/supported_platforms.md

```diff
@@ -0,0 +1,166 @@
+# Supported platforms
+
+## AWS
+
+TODO document this platform
+
+### Deployment types
+
+#### IPI
+
+#### UPI
+
+## vSphere
+
+TODO document this platform
+
+### Deployment types
+
+#### UPI
+
+## Azure
+
+TODO document this platform
+
+### Deployment types
+
+#### UPI
+
+## IBM Cloud
+
+This platform supports deployment of OCP cluster + OCS cluster on top of it.
+
+### Requirements
+
+* Have locally installed ibmcloud CLI, follow [official documentation](https://cloud.ibm.com/docs/cli).
+* Install required plugins for IBM Cloud CLI via command: `ibmcloud plugin install PLUGIN_NAME -f -r 'IBM Cloud'`
+  * infrastructure-service
+  * kubernetes-service
+* Provide credential config file + credential config file described bellow to
+`run-ci` command via `--ocsci-conf` parameters.
+
+### Supported providers
+
+* `vpc-gen2` - This is first implemented supported provider for IBM cloud.
+* `classic` - Not supported yet.
+* `satellite` - Not supported yet.
+
+### Configuration files
+
+* `conf/ocsci/ibm_cloud_cluster_conf.yaml` - this file can be used for deployment
+    on IBM Cloud - this config is for VPC gen 2 provider.
+
+You will need to create also credential file with secret data which should
+never be shared publicly in this repository.
+All needed values are mentioned as placeholders and commented out in
+`conf/ocsci/ibm_cloud_cluster_conf.yaml`
+
+But an example is pasted also here:
+
+```yaml
+# This is the basic config for IBM cloud usage
+---
+ENV_DATA:
+  vpc_id: VPC ID PLACEHOLDER
+  subnet_id: SUBNET ID PLACEHOLDER
+  cos_instance: COS INSTANCE PLACEHOLDER
+AUTH:
+  ibmcloud:
+    api_key: IBM CLOUD API KEY PLACEHOLDER
+    account_id: ACCOUNT ID PLACEHOLDER
+DEPLOYMENT:
+  ocs_secret_dockerconfigjson: BASE64 OF QUAY SECRET PLACEHOLDER
+```
+
+### How to get API key
+
+API key creation is - one time action and is required for run deployment and
+other operation with ibmcloud CLI as we are using login with this API key.
+
+Follow this documentation [access API Key](https://cloud.ibm.com/docs/openshift?topic=openshift-access_cluster#access_api_key).
+
+Command:
+`ibmcloud login -u user@redhat.com -p PASSWORD_PLACEHOLDER -c ACCOUNT_ID_PLACEHOLDER -r us-east`
+
+You should get this info as output where you can get the key:
+
+```
+Please preserve the API key! It cannot be retrieved after it's created.
+
+ID            ApiKey-ID
+Name          my-api-key
+Description
+Created At    2020-10-19T15:05+0000
+API Key       OUR_KEY
+Locked        false
+```
+
+### Deployment types
+
+#### Managed
+
+This deployment type is managed - there is no openshift-installer used for
```

Comment:
Isn't this just UPI? What is the difference between managed and UPI?

Reply:
@clacroix12 the IBM Cloud is not installed by openshift-installer so there is no UPI or IPI kind of deployment. It's completely managed by IBM Cloud. So if you would like to have new machines attached to cluster you interact through IBM Cloud API or CLI and then IBM Cloud will add everything to the cloud.

For more details see here:
https://cloud.ibm.com/docs/openshift?topic=openshift-roks-overview

Basically IBM has some big cluster, it has it's own master nodes and you will get just dedicated space under this big cluster with own worker nodes. You don't have access to the masters but only to the workers.

I think that we can have this managed deployment type used also for OpenShift Dedicated later on. https://www.openshift.com/products/dedicated/